frontend/src/components/characteristics/CharacteristicView.js

```diff
@@ -0,0 +1,184 @@
+  const {enqueueSnackbar} = useSnackbar();
+
+  const userContext = useContext(UserContext);
+  const [state, setState] = useState({
+    loading: true,
+    data: [],
+    selectedUri: null,
+    deleteDialogTitle: '',
+    showDeleteDialog: false,
+  });
+  const [trigger, setTrigger] = useState(true);
+
+  useEffect(() => {
+    if (multi){
+        fetchDataTypes('characteristic').then(res => {
+            if(res.success)
+              setState(state => ({...state, loading: false, data: res.characteristics}));
+          }).catch(e => {
+            reportErrorToBackend(e);
+            setState(state => ({...state, loading: false}))
+            navigate('/dashboard');
+            enqueueSnackbar(e.json?.message || "Error occur", {variant: 'error'});
+          });
+    } else if (single){
+        fetchDataType('characteristic', encodeURIComponent(uri)).then(res => {
+            if(res.success){
+                setState(state => ({...state, loading: false, data: [res.characteristic]}));
+            }
+          }).catch(e => {
+            reportErrorToBackend(e);
+            setState(state => ({...state, loading: false}))
+            navigate('/dashboard');
+            enqueueSnackbar(e.json?.message || "Error occur", {variant: 'error'});
+          });
+    }
+  }, [trigger]);
+
+  const showDeleteDialog = (uri) => {
+    setState(state => ({
+      ...state, selectedUri: uri, showDeleteDialog: true,
+      deleteDialogTitle: 'Delete theme ' + uri + ' ?'
+    }));
+  };
+
+  const handleDelete = async (uri, form) => {
+
+    deleteTheme(uri).then(({success, message})=>{
+      if (success) {
+        setState(state => ({
+          ...state, showDeleteDialog: false,
+        }));
+        setTrigger(!trigger);
+        enqueueSnackbar(message || "Success", {variant: 'success'})
+      }
+    }).catch((e)=>{
+      setState(state => ({
+        ...state, showDeleteDialog: false,
+      }));
+      reportErrorToBackend(e)
+      setTrigger(!trigger);
+      enqueueSnackbar(e.json?.message || "Error occur", {variant: 'error'});
+    });
+
+  };
+
+  const columns = [
+    {
+      label: 'Characteristic Name',
+      body: ({name}) => {
+        return name
+    },
+    sortBy: ({name}) => name
+    },
+
+    {
+      label: 'Characteristic URI',
+      body: ({_uri}) => {
+        if (multi)
+          return <Link colorWithHover to={`/characteristic/${encodeURIComponent(_uri)}/view`}>
+          {_uri}
+        </Link>;
+        if (single)
+          return _uri
+      },
+      sortBy: ({_uri}) => _uri
+    },
+
+    {
+      label: 'Characteristic Code',
+      body: ({codes}) => {
+          return <Link colorWithHover to={`/code/${encodeURIComponent(codes)}/view`}>
+              {codes}
+          </Link>
+      }
+    },
+
+    { 
+      label: 'Value',
+      body: ({value}) => {
+          return value;
+      }
+    },
+
+    { 
+      label: 'Stakeholder Name',
+      body: ({stakeholders}) => {
```

Comment:
It is a list of objects, it is not correct to simply return a list of objects/URI like this, check other dataTypes for example, please. 
Pease check: https://github.com/csse-uoft/CommonApproachSandboxTesting/blob/ef0c296dddb260003aa1db37fc478ded1631c348/frontend/src/components/indicatorReport/IndicatorReportView.js#L62C5-L67C7